src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1379,7 +1387,7 @@ private TypeSymbolWithAnnotations InferResultNullability(BinaryOperatorKind oper
                     return methodOpt.ReturnType;
                 }
             }
-            else if (!operatorKind.IsDynamic() && resultType.IsReferenceType == true)
+            else if (!operatorKind.IsDynamic() && (resultType.IsReferenceType || resultType.IsUnconstrainedTypeParameter()))
```

Comment:
Can this be simplified to `&& !resultType.IsValueType)`? Same comment elsewhere.